react/react_and_the_backend/project_messaging_app.md

```diff
@@ -0,0 +1,19 @@
+### Introduction 
+
+Think of your favorite messaging app - Discord, Telegram, WhatsApp, Facebook Messenger, etc. - and imagine what it would be like if you could build your own. This project will give you a chance to do just that. You'll be building a web app that allows users to send messages to each other. 
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1. Spend some time researching the features your app will need. What will the user interface look like? What will the data model look like? What features will you need to implement? What libraries will you need to use? You don't have to go overboard with features, remember that all those big apps we mentioned earlier have entire teams of developers working on them.
+1. Start building your app by implementing the core features in your backend and frontend. Users should be able to login and send messages to each other.
+1. Once you have that working, you can add profile customization options, group chats, friend list, and other features. It's all up to you.
+1. Deploy your app to the web and showcase it in the submissions below as well as on our Discord. We'd love to see what you've built!
+
+#### Extra Credit
+
+*  Users should be able to send images in chat. 
+*  Users can view who's online.
```

Comment:
Been a while so it might just be a lack of context on my part, but this seems like a lot of "do what you want" without much structure to keep users from getting too caught up in adding features in. Giving users the freedom to build what they want is great, but I do feel like providing some structure as to the minimum they should implement is also good, especially for something that can get as feature heavy as a chat app (multiple chat rooms? sending friend requests? having user profiles? etc)

Reply:
I'm all in for having more clear instructions on what to build. The core features covered in the assignment are:
1. Auth
2. Send messages to other users
3. Profile customization
4. Group chats
5. Friends list

I think these provide a good baseline of project requirements.

As for the extra credits, I feel that's up to the learner on how they want to proceed further. 

Would love your thoughts.

Reply:
This will depend what content users would have learned up to this point so I trust you to swap any of the following suggestions around between extra credit and required, but what if we update the instructions to something like:

```suggestion
1. Keep in mind that the apps we mentioned earlier have entire teams of developers working on them, so you shouldn't feel the need to go overboard with features in your app. At a minimum, though, your app should include the following core functionality:
    1. Authorization
    1. Sending messages to another user
    1. Customizing a user profile
1. Spend some time planning out your app. What will the user interface look like? What will the data model look like? What libraries will you need to use?
1. Start building your app by implementing the core functionality in your backend and frontend.
1. Deploy your app to the web and showcase it in the submissions below as well as on our Discord. We'd love to see what you've built!

#### Extra Credit

*  Allow sending images in chat
*  Add a friends list that users can add other users to and see when someone is online (alternative: "Add a users list to show which user's are currently online"; same thing except maybe a step or two less since it doesn't require adding a friend)
* Allow users to create and send messages in group chats
```

The user profile customization I just kinda threw into the core functionality since it sounded a bit more simplistic but still a nice little additional thing to implement beyond just auth and sending messages. Could argue group chats is more simplistic since the functionality to send messages would already be there.

Also tweaked the verbiage of the extra credits to avoid verbiage like "should" just to make it sound more optional.

Reply:
Thank you for rephrasing the instructions. It sounds more specific, I like it!